software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java

```diff
@@ -251,6 +251,15 @@ public void customize() {
             String saveAsRiakConf = Urls.mergePaths(getRunDir(), "riak.conf");
             DynamicTasks.queueIfPossible(SshEffectorTasks.put(saveAsRiakConf).contents(riakConfTemplate));
             commands.add(sudo("mv " + saveAsRiakConf + " " + getRiakEtcDir()));
+
+            if(!osDetails.isMac()) {
+                if(!com.google.common.base.Strings.isNullOrEmpty(entity.getConfig(RiakNode.RIAK_CONFIG_FILE))) {
```

Comment:
Isn't it better to leave the default to be empty string and check for non-null only here? Any reason empty string should be treated specially?